toolkit/docs/CVE/CVE-Quickstart-Guide.md

```diff
@@ -0,0 +1,167 @@
+# **Document: Fixing CVE Patches for Azure Linux**
+
+This document provides a step-by-step guide to fixing CVE patches for Azure Linux. It includes identifying the CVE, locating and preparing the package, applying the patch, and submitting it for integration into Azure Linux.
+
+---
+
+## **Steps to Fix a CVE Patch**
+
+### **1. Identify the CVE**
+
+1. Navigate to the [Astrolabe Dashboard](https://brave-ocean-0baeae310.5.azurestaticapps.net/#/) to pick a CVE.
+2. Determine the Azure Linux branch for the CVE:
+   - **3.0** → Use the `fasttrack/3.0` branch.
+   - **2.0** → Use the `fasttrack/2.0` branch.
+
+---
+
+### **2. Locate the Package**
+
+1. Clone the **azlinux repository** and switch to the appropriate branch:
+   ```bash
+   git checkout fasttrack/<branch-version>
+   ```
+2. Go to the `SPECS` directory and search for the package:
+   ```bash
+   cd ~/azurelinux/SPECS
+   ls | grep <pkg-name>
+   ```
+
+---
+
+### **3. Retrieve and Prepare the Source**
+
+1. Locate the source URL in the `<pkg>.spec` file:
+   ```bash
+   less ~/azurelinux/SPECS/<pkg>/<pkg>.spec
+   ```
+2. Download the source tarball outside azlinux repo:
+   ```bash
+   wget https://azurelinuxsrcstorage.blob.core.windows.net/sources/<pkg-source>.tar.gz
+   ```
+3. Extract the tarball:
+   ```bash
+   tar -xf <pkg-source>.tar.gz
+   cd <pkg-name>
+   ```
+4. Initialize a Git repository:
+   ```bash
+   git init
+   git add .
+   git commit -m "Initial commit"
+   ```
+
+---
+
+### **4. Apply Existing Patches**
+
+1. Identify existing patches in the `<pkg>.spec` file (e.g., `Patch0`, `Patch1`).
+2. Apply patches sequentially starting from `Patch0`:
+   ```bash
+   patch -p1 --fuzz=0 < ~/azurelinux/SPECS/<pkg>/CVE-xxxx-yyyy.patch
+   ```
+3. Commit the changes:
+   ```bash
+   git add -u
+   git commit -m "In pace with azl"
+   ```
+
+---
+
+### **5. Apply the New CVE Patch**
+
+1. Obtain the CVE patch:
+   - Search on NIST ([NVD](https://nvd.nist.gov/vuln/detail/CVE-xxxx-yyyy)) or GitHub for the patch.
+   - Download and rename it:
+     ```bash
+     wget https://github.com/<pkg>.patch
+     mv <patch-name>.patch CVE-xxxx-yyyy.patch
+     ```
+2. Apply the patch:
+   ```bash
+   patch -p1 --fuzz=0 < CVE-xxxx-yyyy.patch
+   ```
```

Comment:
Can we add "Tips for working with patches that don't apply cleanly"?

https://dev.azure.com/mariner-org/mariner/_wiki/wikis/mariner.wiki/646/Tips-for-working-with-patches-that-don't-apply-cleanly